src/parsers/perf/perfparser.cpp

```diff
@@ -863,24 +890,44 @@ struct PerfParserPrivate
         setParents(&topDownResult.children, nullptr);
     }
 
+    static void buildCallerCalleeResult(const QVector<FrameData>& bottomUpData, FrameData* callerCalleeData)
+    {
+        for (const FrameData& row : bottomUpData) {
+            if (row.children.isEmpty()) {
+                // leaf node found, bubble up the parent chain to add cost for all frames
+                // to the caller/callee data. this is done top-down since we must not count
+                // symbols more than once in the caller-callee data
+                QSet<CallerCalleeLocation> recursionGuard;
+                auto node = &row;
+
+                while (node) {
+                    const auto needle = std::tie(node->symbol, node->binary);
+                    if (!recursionGuard.contains({node->symbol, node->binary})) { // aggregate caller-callee data
+                        auto it = std::lower_bound(callerCalleeData->children.begin(), callerCalleeData->children.end(), needle,
+                            [](const FrameData& frame, const auto needle) { return std::tie(frame.symbol, frame.binary) < needle; });
```

Comment:
you implemented `operator<` above, so use it:

    return CallerCalleeLocation(frame.symbol, frame.binary) < needle;